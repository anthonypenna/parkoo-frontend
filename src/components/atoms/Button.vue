<template>
  <button v-bind="$attrs" :class="`button button--${theme}`">
    <slot />
  </button>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'Button',

  props: {
    theme: {
      default: 'secondary',
      type: String
    }
  }
})
</script>

<style lang="scss" scoped>
.button {
  max-width: 295px;
  width: 100%;
  padding: 14px 0;
  border: none;
  border-radius: 5px;
  font-family: inherit;
  font-size: 16px;
  font-weight: bold;

  &--primary {
    background: #ff9979;
    color: #fff;
  }

  &--secondary {
    background: #f1f1f1;
    color: #000;
  }
}
</style>
