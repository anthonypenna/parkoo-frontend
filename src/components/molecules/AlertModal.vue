<template>
  <Modal>
    <slot />
    <Button @click.native="onConfirm">
      <slot name="button" />
    </Button>
  </Modal>
</template>

<script lang="ts">
import Vue from "vue";
import Modal from "@/components/atoms/Modal.vue";
import Button from "@/components/atoms/Button.vue";

export default Vue.extend({
  name: "AlertModal",
  components: { Modal, Button },
  methods: {
    onConfirm() {
      this.$emit("confirm");
    },
  },
});
</script>
