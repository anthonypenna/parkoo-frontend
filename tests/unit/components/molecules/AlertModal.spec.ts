import { mount, shallowMount } from '@vue/test-utils'
import AlertModal from '@/components/molecules/AlertModal.vue'
import Button from '@/components/atoms/Button.vue'

describe('<AlertModal />', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(AlertModal, {
      slots: { default: 'Hello world!', button: 'Click me' },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('when clicking on the button', () => {
    it('should emit a confirm event', () => {
      const wrapper = mount(AlertModal)
      wrapper.findComponent(Button).trigger('click')
      expect(wrapper.emitted().confirm).toBeTruthy()
    })
  })
})
