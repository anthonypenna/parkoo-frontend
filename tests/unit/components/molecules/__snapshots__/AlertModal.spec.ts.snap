// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertModal /> should render correctly 1`] = `
<modal-stub>Hello world! <button-stub theme="secondary">Click me</button-stub>
</modal-stub>
`;
